File: scripts/migrate_manifest.py
#!/usr/bin/env python3
"""migrate_manifest.py

Ruby スクリプト migrate_manifest.rb の Python 置換。

使い方:
  python scripts/migrate_manifest.py "C:/path/to/1 会社"

依存:
  pip install pyyaml
"""
import sys
from pathlib import Path
import yaml

# 古い変換テーブルは不要になったため削除。
# 処理方針:
# - 入力のキーが `persist_...` の場合は出力を `pr_...` にする
# - 入力のキーが `mf_...` の場合は出力を `pr_...` にする
def migrate_manifests(root: str, yaml_filename: str = '@company.yaml') -> int:
    print("yaml_filename:", yaml_filename)

    # Windows のバックスラッシュをスラッシュに変換
    normalized_root = root.replace('\\', '/')
    pattern = f"{normalized_root}/**/{yaml_filename}"
    print(f"searching for: {pattern}")

    base = Path(normalized_root)
    yaml_file_paths = list(base.rglob(yaml_filename)) if base.exists() else []

    if not yaml_file_paths:
        print(f'no {yaml_filename} found')
        return 1

    for yaml_file_path in yaml_file_paths:
        process_yaml_file(yaml_file_path)

    return 0


def process_yaml_file(yaml_path: Path) -> None:
    try:
        with yaml_path.open('r', encoding='utf-8') as f:
            data = yaml.safe_load(f) or {}
    except Exception as e:
        print(f"failed to read {yaml_path}: {e}")
        data = {}

    mapped = {}
    for src_key, value in data.items():
        if not src_key or value is None:
            continue

        # persist_... -> pr_...
        if src_key.startswith('persist_'):
            suffix = src_key[len('persist_'):]
            dst_key = f'pr_{suffix}'
            mapped[dst_key] = value
            continue

        # mf_... -> pr_...
        if src_key.startswith('mf_'):
            suffix = src_key[len('mf_'):]
            dst_key = f'pr_{suffix}'
            mapped[dst_key] = value
            continue

    # 既定のキーが存在しない場合に空文字列で埋めたい場合はここで追加可能

    try:
        with yaml_path.open('w', encoding='utf-8') as f:
            yaml.dump(mapped, f, allow_unicode=True, sort_keys=False)
        print(f"migrated: {yaml_path}")
    except Exception as e:
        print(f"failed to write {yaml_path}: {e}")


def main(argv):
    if len(argv) < 2:
        print('usage: python migrate_manifest.py "C:/path/to/1 会社" [yaml_filename]')
        return 1

    print("starting migration...", flush=True)
    root = argv[1]
    yaml_filename = argv[2] if len(argv) > 2 else '@company.yaml'
    return migrate_manifests(root, yaml_filename)


if __name__ == '__main__':
    try:
        sys.exit(main(sys.argv) or 0)
    except Exception:
        import traceback
        print('Unhandled exception in migrate_manifest.py:', file=sys.stderr)
        traceback.print_exc(file=sys.stderr)
        sys.exit(1)

File: scripts/migrate_manifest.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'yaml'
require 'pathname'

# キーマッピング定義
PERSIST_KEYS = {
  'persist_long_name' => 'mf_long_name',
  'persist_postal_code' => 'mf_postal_code',
  'persist_address' => 'mf_address',
  'persist_tel' => 'mf_tel',
  'persist_fax' => 'mf_fax',
  'persist_email' => 'mf_email',
  'persist_website' => 'mf_website'
}.freeze

# メイン処理
def migrate_manifests(root)
  # Windowsのバックスラッシュをスラッシュに変換
  normalized_root = root.gsub('\\', '/')
  # @company.yaml ファイルを再帰的に検索
  pattern = File.join(normalized_root, '**', '@company.yaml')
  puts "searching for: #{pattern}"
  company_files = Dir.glob(pattern)

  if company_files.empty?
    puts 'no @company.yaml found'
    return 1
  end

  company_files.each do |company_path|
    process_company_file(company_path)
  end

  0
end

# 個別ファイル処理
def process_company_file(company_path)
  dir_path = File.dirname(company_path)
  manifest_path = File.join(dir_path, '@manifest.yaml')

  # @company.yaml を読み込み
  data = YAML.load_file(company_path) || {}

  # キーをマッピング
  mapped = {}
  PERSIST_KEYS.each do |src_key, dst_key|
    mapped[dst_key] = data[src_key] || ''
  end

  # @manifest.yaml に書き込み
  File.write(manifest_path, YAML.dump(mapped))

  puts "migrated: #{manifest_path}"
end

# エントリーポイント
if __FILE__ == $PROGRAM_NAME
  if ARGV.length < 1
    puts 'usage: ruby migrate_manifest.rb "C:/path/to/1 会社"'
    exit 1
  end

  exit migrate_manifests(ARGV[0])
end
